packages/node-core/src/indexer/postgresBasedMMRDb.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {Db} from '@subql/x-merkle-mountain-range';
+import {Sequelize, DataTypes, Model, ModelStatic} from 'sequelize';
+import {getLogger} from '../logger';
+
+const logger = getLogger('pg-based-db');
+
+interface NodeMap {
+  [key: string]: Buffer;
+}
+
+interface MMRIndexValueStoreAttributes {
+  key: number;
+  value: Buffer;
+}
+
+class MMRIndexValueStoreInstance extends Model<MMRIndexValueStoreAttributes> implements MMRIndexValueStoreAttributes {
+  key!: number;
+  value!: Buffer;
+}
+
+interface LeafLengthAttributes {
+  id: number;
+  value: number;
+}
+
+class LeafLengthInstance extends Model<LeafLengthAttributes> implements LeafLengthAttributes {
+  id!: number;
+  value!: number;
+}
+
+export class PgBasedMMRDB implements Db {
+  private sequelize;
+  private KVStore: ModelStatic<MMRIndexValueStoreInstance>;
+  private LeafLength: ModelStatic<LeafLengthInstance>;
+
+  constructor(sequelize: Sequelize, schema: string) {
+    this.sequelize = sequelize;
+    this.KVStore = this.initKVStoreModel(this.sequelize, schema);
+
+    this.LeafLength = this.initLeafLengthModel(this.sequelize, schema);
+  }
+
+  private initKVStoreModel(sequelize: Sequelize, schema: string): ModelStatic<MMRIndexValueStoreInstance> {
+    MMRIndexValueStoreInstance.init(
+      {
+        key: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+        value: {
+          type: DataTypes.BLOB,
+          allowNull: false,
+        },
+      },
+      {
+        sequelize,
+        modelName: 'KVStore',
+        schema: schema,
+      }
+    );
+    return MMRIndexValueStoreInstance;
+  }
+
+  private initLeafLengthModel(sequelize: Sequelize, schema: string): ModelStatic<LeafLengthInstance> {
```

Comment:
It seems like overkill to have a table for a single value. It probably makes more sense to use the metadata table or use a specific key in the other table (maybe -1 as the key).